### PR TITLE
Feature/07062020 get lichess pgn

### PR DESCRIPTION
**Pull request by jorgedclopes**

### CODE REVIEW

File: src/load_games.py [hunk 1]
```diff
@@ -21,38 +64,47 @@ def load_games(path: str = 'resources/PGN_database',
...
-            Default: None
-
     Returns
     -------
         List[dict]
             list of games in DB, sorted by date.
 
     """
 
+    fname = []
```
Comment (duartenina) [inline src/load_games.py]: Replace flatten of two maps for single list comprehension.

```suggestion
    fname = [os.path.join(folder, file) for folder, children, files in os.walk(base_folder) for file in files]
```

File: src/load_games.py [hunk 2]
```diff
@@ -21,38 +64,47 @@ def load_games(path: str = 'resources/PGN_database',
...
+    elif os.path.exists(path):
+        fname = [path]
+
     games = list()
-    for file in os.listdir(path):
-        with open(os.path.join(path,
-                               file), 'r') as f:
-            ind_game = ast.literal_eval(f.read())
-            games.append(ind_game)
+    for file in fname:
```
Comment (duartenina) [inline src/load_games.py]: `file` is already used by Python.

```suggestion
    for filename in filenames:
```

**Comment by duartenina** [comment]
.

File: src/get_games_chesscom.py [hunk 3]
```diff
@@ -15,6 +17,21 @@ def get_chess_dot_com_games(
     path: str = "resources/",
     new_file_name: str = 'chessdotcom_games.pgn'
 ):
+    """Flattens a list completely.
```
Comment (duartenina) [inline src/get_games_chesscom.py:20]: Wrong comment?

File: src/load_games.py [hunk 4]
```diff
@@ -5,14 +5,57 @@
...
+        arg_list : list[list[...]]
+            nested list
+
+    Returns
+    -------
+        list[...]
+            flattened list of games in list.
+
+    """
+    return [item for sublist in arg_list
```
Comment (duartenina) [inline src/load_games.py:30]: It should be clear in the documentation this function only transforms 2-level lists in 1-level lists.

File: src/load_games.py [hunk 5]
```diff
@@ -5,14 +5,57 @@
...
+                flattened dict of games in list.
+
+    """
+    items = []
+    for k, v in arg_dict.items():
+        new_key = k
+        if isinstance(v, collections.MutableMapping):
+            items.extend(flatten_dict(v).items())
+        else:
+            items.append((new_key, v))
```
Comment (duartenina) [inline src/load_games.py:54]: What happens (should happen?) when new_key already exists? Current code only uses the last key in the list.

**Comment by jorgedclopes** [comment]
Abandoned development.
Chosen to follow another way.
Put the information into a class to be analysed.

**Pull request closed without merging**